### ISSUE TITLE
[Development] Add Pytorch and Keras to Jupyterhub

### ISSUE THREAD

**Issue by harshpavuluri** [ironhacks/ironhub#10]
Summary:

Per Professor Sun's request, we add Pytorch and Keras to the libraries installed on the Jupyterhub. Also add these to the approved list of libraries

Tasks:
- [ ] Add Pytorch and Keras to the Docker image and redeploy
- [x] Update the approved list of libraries for the participants

**Comment by jialincheoh** [comment]
approved list of libraries been updated for the participants @harshpavuluri @sbrunswi , pending @ManthanKeim installing pytorch and keras on ironhub

**Comment by ManthanKeim** [comment]
I don't think I'm going to mess up with the hub this close to the hack, installing this libraries mean me creating the jupyterhub image from scratch, but I can always install them when we're not in a cruical situation like this, partcipants can do a pip install directly and they'll be able to access them anyways.

What we can do is make a list of all the libraries and in the long term, I would install them for once after the hack and we don't have to create the image again.

**Comment by ManthanKeim** [comment]
This will be installed in the next update of Ironhub, request more library if needed,

https://github.com/ironhacks/ironhub/blob/master/libraries_need_to_be_installed

**Comment by sbrunswi** [comment]
@crystalhappycat  and @aronlindberg: Please tell me what else is needed.

**Comment by sbrunswi** [comment]
So we need Tensorflow. I will add a list of DL libraries in a bit.

**Comment by harshpavuluri** [comment]
alembic                       1.8.1
altair                        4.2.0
anyio                         3.6.1
argon2-cffi                   21.3.0
argon2-cffi-bindings          21.2.0
asttokens                     2.0.8
async-generator               1.10
attrs                         22.1.0
Babel                         2.10.3
backcall                      0.2.0
backports.functools-lru-cache 1.6.4
beautifulsoup4                4.11.1
bleach                        5.0.1
blinker                       1.5
bokeh                         2.4.3
Bottleneck                    1.3.5
brotlipy                      0.7.0
cached-property               1.5.2
cachetools                    5.2.0
certifi                       2022.9.24
certipy                       0.1.3
cffi                          1.15.1
charset-normalizer            2.1.1
click                         8.1.3
cloudpickle                   2.2.0
colorama                      0.4.5
conda                         22.9.0
conda-package-handling        1.9.0
contourpy                     1.0.5
cryptography                  38.0.2
cycler                        0.11.0
Cython                        0.29.32
cytoolz                       0.12.0
dask                          2022.10.0
debugpy                       1.6.3
decorator                     5.1.1
defusedxml                    0.7.1
dill                          0.3.5.1
distributed                   2022.10.0
entrypoints                   0.4
et-xmlfile                    1.0.1
executing                     1.1.1
fastjsonschema                2.16.2
flit_core                     3.7.1
fonttools                     4.37.4
fsspec                        2022.8.2
gmpy2                         2.1.2
google-api-core               2.10.2
google-auth                   2.13.0
google-cloud-bigquery         3.3.5
google-cloud-bigquery-storage 2.16.2
google-cloud-core             2.3.2
google-crc32c                 1.5.0
google-resumable-media        2.4.0
googleapis-common-protos      1.56.4
greenlet                      1.1.3
grpcio                        1.50.0
grpcio-status                 1.50.0
h5py                          3.7.0
HeapDict                      1.0.1
idna                          3.4
imagecodecs                   2022.9.26
imageio                       2.22.0
importlib-metadata            4.11.4
importlib-resources           5.10.0
ipykernel                     6.16.0
ipympl                        0.9.2
ipython                       8.5.0
ipython-genutils              0.2.0
ipywidgets                    8.0.2
jedi                          0.18.1
Jinja2                        3.1.2
joblib                        1.2.0
json5                         0.9.5
jsonschema                    4.16.0
jupyter-client                7.3.4
jupyter_core                  4.11.1
jupyter-server                1.21.0
jupyter-telemetry             0.1.0
jupyterhub                    3.0.0
jupyterlab                    3.4.8
jupyterlab-pygments           0.2.2
jupyterlab_server             2.16.0
jupyterlab-widgets            3.0.3
kiwisolver                    1.4.4
libmambapy                    0.27.0
llvmlite                      0.39.1
locket                        1.0.0
lz4                           4.0.0
Mako                          1.2.3
mamba                         0.27.0
MarkupSafe                    2.1.1
matplotlib                    3.6.1
matplotlib-inline             0.1.6
mistune                       2.0.4
mpmath                        1.2.1
msgpack                       1.0.4
munkres                       1.1.4
nbclassic                     0.4.5
nbclient                      0.7.0
nbconvert                     7.2.1
nbformat                      5.7.0
nest-asyncio                  1.5.6
networkx                      2.8.7
notebook                      6.5.1
notebook-shim                 0.1.0
numba                         0.56.3
numexpr                       2.8.3
numpy                         1.23.3
oauthlib                      3.2.1
openpyxl                      3.0.10
packaging                     21.3
pamela                        1.0.0
pandas                        1.5.0
pandocfilters                 1.5.0
parso                         0.8.3
partd                         1.3.0
patsy                         0.5.3
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        9.2.0
pip                           22.3
pkgutil_resolve_name          1.3.10
prometheus-client             0.15.0
prompt-toolkit                3.0.31
proto-plus                    1.22.1
protobuf                      4.21.7
psutil                        5.9.2
ptyprocess                    0.7.0
pure-eval                     0.2.2
pyarrow                       9.0.0
pyasn1                        0.4.8
pyasn1-modules                0.2.8
pycosat                       0.6.3
pycparser                     2.21
pycurl                        7.45.1
Pygments                      2.13.0
PyJWT                         2.5.0
pyOpenSSL                     22.1.0
pyparsing                     3.0.9
pyrsistent                    0.18.1
PySocks                       1.7.1
python-dateutil               2.8.2
python-json-logger            2.0.1
pytz                          2022.4
pytz-deprecation-shim         0.1.0.post0
PyWavelets                    1.3.0
PyYAML                        6.0
pyzmq                         24.0.1
requests                      2.28.1
rpy2                          3.5.1
rsa                           4.9
ruamel.yaml                   0.17.21
ruamel.yaml.clib              0.2.6
ruamel-yaml-conda             0.15.80
scikit-image                  0.19.3
scikit-learn                  1.1.2
scipy                         1.9.1
seaborn                       0.12.0
Send2Trash                    1.8.0
setuptools                    65.5.0
simplegeneric                 0.8.1
simpy                         4.0.1
six                           1.16.0
sniffio                       1.3.0
sortedcontainers              2.4.0
soupsieve                     2.3.2.post1
SQLAlchemy                    1.4.42
stack-data                    0.5.1
statsmodels                   0.13.2
sympy                         1.11.1
tables                        3.7.0
tblib                         1.7.0
terminado                     0.16.0
threadpoolctl                 3.1.0
tifffile                      2022.10.10
tinycss2                      1.1.1
tomli                         2.0.1
toolz                         0.12.0
tornado                       6.1
tqdm                          4.64.1
traitlets                     5.4.0
typing_extensions             4.4.0
tzdata                        2022.5
tzlocal                       4.2
unicodedata2                  14.0.0
urllib3                       1.26.11
wcwidth                       0.2.5
webencodings                  0.5.1
websocket-client              1.4.1
wheel                         0.37.1
widgetsnbextension            4.0.3
xlrd                          2.0.1
zict                          2.2.0
zipp                          3.9.0


Current libraries on the hub (includes dependencies)

**Comment by sbrunswi** [comment]
1) keras
2) pytorch
3) tensorflow
4) Theano

those ones we need. We need to update sheet then.

**Comment by sbrunswi** [comment]
https://docs.google.com/spreadsheets/d/1iIbhE5Px3mb2qdqRlnsdEUtnRL90oxm73YqMoSCXeXw/edit#gid=686999167

**Comment by sbrunswi** [comment]
Solution: Instructions as a forum posts on how to actually install keras, pytorch, tensorflow - I pointed in the task already to it and made a note in the list of libraries. @harshpavuluri we emphasize it in tutorial 6a! prepare the submission because there is a code snippet - we could just do a forum post plus a small tutorial 7: How to install pytorch, keras, and tensorflow!

**Comment by sbrunswi** [comment]
@crystalhappycat - could you write a forum post about that?

**Comment by crystalhappycat** [comment]
There's no tutorial 7. If you give me instruction for installation, I can make a forum post.

**Comment by sbrunswi** [comment]
It isn the task statement! Just check it out.

**Comment by jialincheoh** [comment]
Tutorial 7 still pending final approval @sbrunswi  @crystalhappycat
